### PR TITLE
profiling: make a draft for discussion

### PR DESCRIPTION
**Pull request by tai271828**
An easy script to show what I know after reading the source of modmesh about how to use the following items:
- manipulate the euler1d solver of modmesh and create a 1D shock tube problem based on the solver.
- profiling the problem by using profile.hpp. That is to say, a stop watch in this code change.

I am looking forward to more discussion or suggestions that help us make #135 move forward and keep us on the same page. For example, is this what you expect about the approach of profiling?

### CODE REVIEW

File: contrib/profiling.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    gamma=1.4,
+    pressure1=1.0,
+    density1=1.0,
+    pressure5=0.1,
+    density5=0.125,
+)
+
+shock_tube.build_numerical(xmin=-1, xmax=1, ncoord=21, time_increment=0.05)
+
+# reset lap
```
Comment (yungyuc) [inline contrib/profiling.py:16]: Is there a report or something for the test case?

Comment (tai271828) [reply contrib/profiling.py:16]: No. We can (and should) implement that for sure. 

Do we want to add profiling a part of test case? If yes, then we may want to have a benchmark that is agreed on mutually.

Comment (yungyuc) [reply contrib/profiling.py:16]: Before making automatic tests for runtime, perhaps we can first work on the (automatic generation of the) profiling reports.  The reports are what we want from a profiler.  One example of the reporting is https://github.com/yse/easy_profiler .

However here I think we should first develop a [scoped-based profiler](https://yyc.solvcon.net/en/latest/nsd/11arraydesign/arraydesign.html?highlight=scope#scoped-based-timer), which is usually ad hoc and does not have a lot of simple examples online.  We may use the examples of the sample-based profiling, but start with a scope-based one.

Comment (tai271828) [reply contrib/profiling.py:16]: Thanks for the comments. The roadmap sounds good to me. Let me have a look of  https://github.com/yse/easy_profiler

**Comment by tai271828** [comment]
Close this PR since we now have this driving script from gist [euler1d_profiling-py](https://gist.github.com/tai271828/d5f50e4a06274b1203ed79f6739039dd#file-euler1d_profiling-py).

This driving script runs through a real shock tube problem and the result is easier to ready.

**Pull request closed without merging**